chat.go

```diff
@@ -157,6 +158,11 @@ func (api *Client) SendMessageContext(ctx context.Context, channelID string, opt
 		return "", "", "", err
 	}
 
+	//Check slack response for error
+	if response.SlackResponse.Ok == false {
+		return "", "", "", fmt.Errorf("slack API error: %s", response.SlackResponse.Error)
```

Comment:
two things:
1. lets return the error string directly. don't prepend 'slack API error:' to it.
2. lets make this a function on SlackResponse.

```
type SlackResponse struct {
	Ok    bool   `json:"ok"`
	Error string `json:"error"`
}

func (t SlackResponse) Err() error {
  if t.Ok {
    return nil
  }

  return errors.New(t.Error)
}
```